handlers/istio_config.go

```diff
@@ -29,7 +41,33 @@ func IstioConfigList(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	var istioConfigValidations models.IstioValidations
+
+	wg := sync.WaitGroup{}
+	if includeValidations {
+		wg.Add(1)
+		go func(namespace string, istioConfigValidations *models.IstioValidations, err *error) {
+			defer wg.Done()
+			// We don't filter by objects when calling validations, because certain validations require fetching all types to get the correct errors
+			istioConfigValidationResults, errValidations := business.Validations.GetValidations(namespace, "")
+			if errValidations != nil && err == nil {
+				*err = errValidations
+			} else {
+				if len(parsedTypes) > 0 {
+					istioConfigValidationResults = istioConfigValidationResults.FilterByTypes(parsedTypes)
+				}
+				*istioConfigValidations = istioConfigValidationResults
+			}
+		}(namespace, &istioConfigValidations, &err)
+	}
+
 	istioConfig, err := business.IstioConfig.GetIstioConfigList(criteria)
```

Comment:
Perhaps the "GetIstioConfigList(criteria)" can be before the if and we can merge both "if includeValidations" into one ?

Reply:
Well, reading twice it sounds correct, as we are not using a Add(2) in the wait, the async call should go first.

Reply:
Yep, the GetIstioConfigList can block this processing goroutine - there's no need to create one more and then block this one to wait for that extra.